lib/macho/load_commands.rb

```diff
@@ -741,6 +744,42 @@ def initialize(view, cmd, cmdsize, htoffset, nhints)
       super(view, cmd, cmdsize)
       @htoffset = htoffset
       @nhints = nhints
+      @table = TwolevelHintsTable.new(view, htoffset, nhints)
+    end
+
+    # A representation of the two-level namespace lookup hints table exposed
+    # by a {TwolevelHintsCommand} (`LC_TWOLEVEL_HINTS`).
+    class TwolevelHintsTable
+      # @return [Array<MachO::TwoLevelHintsTable::TwoLevelHint>] all hints in the table
+      attr_reader :hints
+
+      # @param view [MachO::MachOView] the view into the current Mach-O
+      # @param htoffset [Fixnum] the offset of the hints table
+      # @param nhints [Fixnum] the number of two-level hints in the table
+      # @api private
+      def initialize(view, htoffset, nhints)
+        format = (view.endianness == :little) ? "V<" : "V>"
```

Comment:
Can we re-use `MachOStructure.specialize_format`, so that we can avoid specifying both little- and big-endian cases over and over again? Keeping this logic mostly in a single location will also make it easier to restore Ruby 1.8.7 compatibility, if we have to (though I certainly hope we can avoid that). Maybe it's worth moving `specialize_format` somewhere else to make it more generally available …